### DISCUSSION TITLE
Fat 32 limits us to 32GB.  Is there a way to increase this?

### DISCUSSION THREAD

**Discussion by idaband** [gtxaspec/wz_mini_hacks#90] [Q&A]
So... are we limited to only using 32GB cards?  if I used a large card can I use the second partition somehow for recording videos?

Thank you in advanced.

**Comment by claudobahn** [comment]
You can go larger, see here: https://github.com/gtxaspec/wz_mini_hacks/discussions/83#discussioncomment-2948999

**Reply by virmaior** [reply]
That's interesting. I didn't have success with using larger SD cards that work post boot, because there's a limitation on the boot loader. 
I guess the key is making sure they are FAT32 formatted which is somehow offspec for > 32 GB ?

**Comment by endertable** [comment]
I use a small utility (in WIndows) named guiformat-x64.exe or guiformat.exe and formatted up to 64GB.  Haven't tried anything larger so can't confirm nor deny larger sizes.

**Comment by virmaior** [comment]
for OSX,

```
diskutil list

```

identify the correct volume, then


````
diskutil partitionDisk /dev/disk4 1 MBRFormat "MS-DOS FAT32" NONAME R

````

and there you go.

**Comment by skilau** [comment]
I used the above mentioned "guiformat-x64.exe" to format my 128G card to FAT32.
It works fine, and all 128G is seen just fine in all my 5 v3 Cams.

PS: I know we have to stay with FAT32 because of the existing bootloader (that we hijack) requires it.

But it is painful at times, because doing a format of the card in the Wyze app formats us to exFAT, which is a pain to reverse sometimes.

(In fact, when I left it at exFAT by mistake once, copied all the wz_mini_hacks file over, it ignores the files, and goes directly into the regular Wyze firmware... Took me a few minutes to figure out what went wrong!)

**Comment by UncleJohnsBnd** [comment]
I am wondering what idaband was wondering(at least I think we are on same thought)  can you take a 256gb sd card and create like a 200mb fat32 partition for the hack and then use extfat on the other partition for the recordings.

I can get by with 128gb if needed and just do the whole sd card as fat32 but would prefer to have the extra space to have more recording days(I always record in HD) of history.

**Reply by skilau** [reply]
Agreed.  I am hoping that we will be able to do it this way eventually as well.
I think it was mentioned that once the OverlayFS is brought in, it may make doing this easier.

**Reply by virmaior** [reply]
that's an interesting question, but you probably don't actually need two partitions to solve this.

1. You need a FAT32 formatted microSD card for the hack to load because the hack gets loaded by the bootloader (I'm actually not sure what it will do if you've got multiple partitions), and the bootloader doesn't know how to read exfat
2. The limit on FAT32 is not actually 32gb. It just happens to be the case that windows' standard formatter won't do it, because it thinks exFAT is better at larger sizes  (see for instance https://www.wikihow.com/Format-FAT32 noting you'd need a third party app for it; for windows users, see above).
3. If you format a card inside of wyze app, it will make it exfat, because they added exfat support at around 4.36.2.5 or so.

Even if the bootloader can decode a flash drive with 2 partitions on it, the current set up is going to expect that files get saved to /media/mmc/record/ which is going to be where it mounted the first partition.
I guess you could add a cron job or something to sweep them onto the second partition, but this seems way more complicated than just formatting a larger SD card with FAT32

**Reply by virmaior** [reply]
@skilau  not saying you're wrong but what's the benefit of using two partitions to accomplish this over the one?

**Reply by UncleJohnsBnd** [reply]
So that windows util will in-fact let you format 256gb as fat32?  If so,  then really this is a non-issue for me.

I had a bad understanding that 128gb was max for fat32.

**Reply by skilau** [reply]
@virmaior Minor reasons, to be honest.

First, since the only reason we need FAT32 at all, is because the bootloader needs to read the kernel out from flash, and it must be on FAT32, and the size of that kernel is well bounded, (2-3M at most), so it seems logical to just have a small FAT32 partition holding that image, and then partition the rest as either exFAT or perhaps just go directly to ext4 or something.

Second, since Wyze eats up all space on the card, and then starts deleting old for new, I don't like the fact that they might take up all the space on the disk, without leaving us space for logs, etc.
We could regulate this by creating another exFAT partition and mount that specifically for video output from Wyze.

Of course, the downside of this, is that dragging and dropping the files like we do today won't work anymore.
It would have to be done with an external app like Balena Etcher, or similar.

However, I suspect the convenience of the drag-n-drop is so important/easy, that the project probably should just keep it the way it is today.

**Reply by UncleJohnsBnd** [reply]
Did a little research and just to get in print the max partition size for fat32 is 2TB.

Still think the dual partition option provides best architecture though.... If it could work.

**Reply by guarddog13** [reply]
Use the raspi imager it natively formats any size card to fat32
